_posts/2020-06-07-higher-order-hydras.markdown

```diff
@@ -0,0 +1,491 @@
+The game ends when the hydra is reduced to a single node.
+
+To obtain a fast-growing function, we can fix \\\(N\\\), say, \\\(N = 1\\\) at the first step, then \\\(N = 2\\\), \\\(N = 3\\\), and so on,
+and decide on a simple rule for where to cut, say, always choosing the rightmost leaf.
+Then, \\\(Hydra(k)\\\) is the number of steps needed for the game to end starting with a path of length \\\(k\\\), that is, a linear stack of \\\(k + 1\\\) nodes.
+
+Buchholz hydras extend Kirby-Paris hydras with labels, and require a few extra transformation cases.
+\\\(0\\\)-labelled nodes behave the same as in the Kirby-Paris hydra game.
+Integers greater than \\\(0\\\) turn into their predecessor but make the tree deeper.
+\\\(ω\\\), rather than being chopped, turns into the current value of the step counter, more or less.
+I won't explain Buchholz hydras in detail here, but you can read more about them yourself.
+
+Googology wiki has fine articles on
+[Kirby-Paris hydras](https://googology.wikia.org/wiki/Kirby-Paris_hydra) and
+[Buchholz hydras](https://googology.wikia.org/wiki/Buchholz_hydra),
+which I suggest you read before coming back here.
+
+In this post, I introduce a higher extension of these hydras, which I believe is interesting in its own right.
+
+</div>
+
+# An idea to nest hydras
+{: #nested-hydras}
+
+{% include collapser.markdown %}
+
+<div>
+
+In exploring ways to build faster growing functions than the one based on Buchholz hydras, a natural question to ask is "what if we used higher ordinals as labels?"
+and then since hydras are ordinal-like, "what if we used nested hydras as labels?"
+We can skip the details of adapting ordinal transformation rules for hydras right now, but in any case, these nested hydras are surely more powerful than plain Buchholz hydras.
+
+I then thought to add a singleton, which I'll call \\\( \lambda \\\), which when encountered, expands into a hydra involving labels which are hydras involving labels which are hydras... \\\(N\\\) layers deep, containing no mention of \\\( \lambda \\\).
+This is, in some sense, a fixed point ordinal of hydra nesting, or for accuracy, the first common fixed point of whatever ordinal operation nesting hydras corresponds to.
+I cannot think of what ordinal such a system corresponds to, but surely it is stronger than without \\\( \lambda \\\).
+Why stop there?
+We can rename \\\( \lambda \\\) to \\\( \lambda_0 \\\), then introduce \\\( \lambda_1 \\\), the second common fixed point of hydra nesting, which expands like \\\( \lambda_0 \\\), except the nested hydras contain \\\( \lambda_0 \\\).
+Why stop there?
+We can make subscripts ordinals, or even hydras.
+Then we can add another layer, say, \\\( \xi_0 \\\), the first common fixed point of \\\( \lambda \\\) nesting, which expands into a stack of nested \\\( \lambda \\\) hydras, and so on.
+
+</div>
+
+# Visualizing nested hydras
+{: #visualizing-nested-hydras}
```

Comment:
😮 What do you use to make your images? They're really cool!

Reply:
GIMP. Honestly I should have probably drawn pretty diagrams in Inkscape or auto-generated them with graphviz, but eh, this is what I ended up using. And thank you.